Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUtils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        namespace PrefabUndoUtils
```

Comment:
I found this class pretty helpful. Thanks for adding this. One thing I notice is you can rename Generate* to Capture* since the function is appending the patch to array.

Reply:
`GenerateUpdateEntityPatch` actually creates an array of changes in given patch so I will keep using this name, but we can change the other two functions' name to `AppendAddEntityPatch` and `AppendRemoveEntityPatch`.